liam2/entities.py

```diff
@@ -107,9 +107,12 @@ def from_def(cls, name, fielddef):
                 output = fielddef.pop('output', True)
                 default_value = fielddef.pop('default', default_value_by_strtype[strtype])
                 if fielddef:
-                    invalid_keywords = sorted(fielddef.keys())
-                    raise SyntaxError('invalid keyword(s) found in field definition: %s'
-                                      % ', '.join(invalid_keywords))
+                    invalid_keywords = [str(definition) for definition in sorted(fielddef.keys())]
```

Comment:
AFAICT, this line change is not needed, given the switch to .format below which will not break on non-string arguments.

Reply:
Ok I will test that.

Reply:
Actually it is the join that breaks if provided with a boolean. So the stringification is needed.

Reply:
Oh, sorry, I missed the join.